src/libsystemd-network/sd-radv.c

```diff
@@ -357,22 +370,12 @@ static int radv_timeout(sd_event_source *s, uint64_t usec, void *userdata) {
         timeout = min_timeout + random_u64_range(max_timeout - min_timeout);
         log_radv(ra, "Next Router Advertisement in %s", FORMAT_TIMESPAN(timeout, USEC_PER_SEC));
 
-        r = event_reset_time(ra->event, &ra->timeout_event_source,
-                             CLOCK_BOOTTIME,
-                             usec_add(time_now, timeout), MSEC_PER_SEC,
-                             radv_timeout, ra,
-                             ra->event_priority, "radv-timeout", true);
-        if (r < 0)
-                goto fail;
-
-        ra->ra_sent++;
-
-        return 0;
-
-fail:
-        sd_radv_stop(ra);
-
-        return 0;
+        return event_reset_time(
+                        ra->event, &ra->timeout_event_source,
+                        CLOCK_BOOTTIME,
+                        usec_add(time_now, timeout), MSEC_PER_SEC,
+                        radv_timeout, ra,
+                        ra->event_priority, "radv-timeout", true);
```

Comment:
Looks like it works in the sense that networkd replies to router solicitations with unspecified source addresses and resets the timer:
```sh
02:20:11.354945 IP6 :: > ff02::2: ICMP6, router solicitation, length 8
02:20:11.355525 IP6 H > ff02::1: ICMP6, router advertisement, length 64
```
```sh
systemd-networkd[166190]: ve-TEST-01: RADV: Next Router Advertisement in 4min 10s
```
but it seems it no longer logs that it sends anything. With unicast routers advertisements it logs something like
```sh
RADV: Sent solicited Router Advertisement to fe80::3430:b6ff:fe30:94d7
```

Reply:
Log message is updated.